Rxjs testing

KorinaGraber opened the pull request:
After some investigation, I've discovered that the reason scheduling rxjs tests is currently not possible is that the scheduler always depends on the current browser time. If we instead tie this to the elapsed time of the fake async zone based on 'tick', we'll gain the ability to also schedule rxjs tasks in tests. Requires wrapping angular 'tick' so we can track the elapsed time.

This only impacts unit tests, so it shouldn't need to be incorporated into the WIP pipeline.

https://renovo.myjetbrains.com/youtrack/issue/RLv21-54

Commit:
Added tests for the fakeAsync wrapper behavior

Test coverage

Commit:
Allow for scheduling rxjs tasks with fakeAsync

After some investigation, I've discovered that the reason scheduling rxjs tests is currently not possible is that the scheduler always depends on the current browser time. If we instead tie this to the elapsed time of the fake async zone based on 'tick', we'll gain the ability to also schedule rxjs tasks in tests. Requires wrapping angular 'tick' so we can track the elapsed time.

Commit:
Drop unused comment

Cleanup

Commit:
Save the existing scheduler 'now' and reset it after the fakeAsync run

This further prevents any chance that these changes will have an impact on the application or other unit tests.

JoshDSommer commented:
I would like to see the name of this changed to rlFakeAsync or something of that nature. For no reason other than to prevent any confusion for future devs between this and Angulars FakeAsync.

Other than that :shipit:

Commit:
Prefix custom async test functionality with 'rl'

This change was recommended to ensure that devs can clearly differentiate these from the native angular `fakeAsync` and `tick`. For now, exporting the original names as aliases for backwards compatibility.

KorinaGraber commented:
How about this? https://github.com/RenovoSolutions/TypeScript-Angular-Utilities/pull/161/commits/978a8de7bfe03c349b48a4b086a5f1d2194d5fb4

JoshDSommer commented:
Sounds good to me :dancer:
